twtr/src/integration-test/groovy/twtr/AccountIntegration2Spec.groovy

```diff
@@ -0,0 +1,106 @@
+package twtr
+
+
+import grails.test.mixin.integration.Integration
+import grails.transaction.*
+import spock.lang.*
+
+@Integration
+@Rollback
+class AccountIntegration2Spec extends Specification {
+
+    final static String goodHandle = "@scsu-huskies"
+    final static String goodEmail  = "testemail@test.com"
+    final static String goodPassword = "abc123ABC"
+    final static String goodDisplayName = "SCSU Huskies"
+
+    final static String handle1 = "@scsu-huskies1"
+    final static String email1  = "testemail1@test.com"
+    final static String password1 = "abc123ABC"
+    final static String displayName1 = "SCSU Huskies1"
+
+    final static String handle2 = "@scsu-huskies2"
+    final static String email2  = "testemail2@test.com"
+    final static String password2 = "abc123ABC"
+    final static String displayName2 = "SCSU Huskies2"
+
+    def setup() {
+    }
+
+    def cleanup() {
+    }
+
+    void "saving an account with a non-unique email will fail."() {
+        setup: "Add the first account with email ${goodEmail}"
+        def account1 = new Account(handle: goodHandle, emailAddress: goodEmail, password: goodPassword, displayName: goodDisplayName)
+        account1.save(flush: true)
+
+        when: "Another account with ${goodEmail} is attempted to be saved."
+        def sus = new Account(handle: handle1, emailAddress: goodEmail, password: password1, displayName: displayName1)
+        sus.save(flush: true)
+
+        then: "The 2nd account should be invalid and only the 1st account should exist."
+        !sus.validate()
+        Account.get(account1.id).handle == goodHandle
+        !Account.findByHandle(sus.handle)
+        Account.findAllByEmailAddress(goodEmail).size == 1
+        Account.findByEmailAddress(goodEmail).handle == goodHandle
+
+        cleanup:
+        Account.list().each {it.delete(flush: true, failOnError: true) }
+    }
+
+    void "two accounts may follow each other"() {
+        setup:
+        def sus1 = new Account(handle: handle1, emailAddress: email1, password: password1, displayName: displayName1)
+        sus1.save(flush: true)
+
+        def sus2 = new Account(handle: handle2, emailAddress: email2, password: password2, displayName: displayName2)
+        sus2.save(flush: true)
+
+        when: 'sus2 starts following sus1'
+        sus1.addToFollowing(sus2)
+        sus2.addToFollowers(sus1)
+        sus1.save()
```

Comment:
It doesn't matter too much, either way.  I suppose I could change it to be flush: true as well as failOnError: true.  Mostly, since the save is happening right after, i figured I'd put them in one flush.